src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,264 @@
+
+``` C++
+pub.publish(msg);
+```
+Publish the message using the previously initialized Publisher `pub`.
+
+``` C++
+ros::spinOnce();
+```
+There is also the function `ros::spin()`. The main difference is that `ros::spinOnce()` is used when other lines of code, and processing arriving messages, need to be executed along side each other, such as in a while loop. Whereas `ros::spin()` should be called at the end of a file.
+
+### Python
+
+A basic setup for ROS topics in Python would look something like this:
+
+``` Python
+rospy.init_node('Demo', anonymous=True)
+rospy.Subscriber('plot', Float64, callback)
+pub = rospy.Publisher('chatter', String, queue_size=QUEUE_SIZE)
+```
+
+To break this down,
+
+``` Python
+rospy.init_node('Demo', anonymous=True)
+```
+Initialize ROS and specify the name of the node (which is the name of the file in this case).
+
+``` Python
+rospy.Subscriber('plot', Float64, callback)
+```
+Initialize the Subscriber. State the name of the topic (`plot`), the message type we will be publishing (float, which is actually in the class std_msgs.msg.Float64), and the function that will be called when a message is received (callBack(data)).
+
+``` Python
+pub = rospy.Publisher('chatter', String, queue_size=QUEUE_SIZE)
+```
+Initialize the Publisher. State the name of the topic (`chatter`), the message type we will be publishing (String, which is actually in the class std_msgs.msg.String), and the queue size (1). The queue size is the maximum number of messages kept as a buffer before throwing away old ones if we are publishing too quickly.
+
+The callback function for a Subscriber looks something like this:
+
+``` Python
+def callback(data: Float64) -> None:
+    global pi 
+    pi = data.data
+```
+This function will get called when a new message has arrived on the topic `plot`. The message received by the Subscriber, sent by the Publisher, is denoted as `msg`.
+
+Publishing a message on a topic would look something like this:
+
+``` Python
+pi_str = "currently at %s" % pi
+pub.publish(hello_str)
+```
+
+To break this down,
+
+``` Python
+pi_str = "currently at %s" % pi
+```
+Create the message.
+
+``` Python
+pub.publish(hello_str)
+```
+Publish the message using the previously initialized Publisher `pub`.
+
+If you have glanced at the C++ example code, you may have noticed the function `ros::spinOnce()` and its differences with `ros::spin()`. Python only has `rospy.spin()`, which can be completely replaced by a for loop, which is why you do not see it in the Python example code.
+
+## Running a Publisher and Subscriber
+Before you are able to run the example code, find the CMakeLists.txt file in the directory /workspaces/WRoverPlayground/src/topic_training. This file will include the the configurations, builds, and installs required to run your program as well as the files you will be running. Do not change anything in the file as it should already be set up for the demo files to work. 
```

Comment:
Same as before, adding `code` can make the path more readable and stand out. 

Also, it might be nice to highlight that they don't really need to understand `CMakeLists` as it's pretty complicated and I can imagine new members quitting because of how daunting it seems. 

```suggestion
Before you are able to run the example code, find the CMakeLists.txt file in the directory `/workspaces/WRoverPlayground/src/topic_training`. This file will include the the configurations, builds, and installs required to run your program as well as the files you will be running. Do not change anything in the file as it should already be set up for the demo files to work. You also don't need to understand anything in the file, as it's way beyond the scope of this training. 
```

Reply:
Fixed in 5182f6b.